File: PN11.cpp


#include<iostream>
using namespace std;

class Number
{
    private:
        int iNo;        

    public:
        
        void Accept()       
        {
            cout<<"Enter the value : "<<endl;
            cin>>this->iNo;
        }

        void Display()      
        {
            cout<<"Value is : "<<this->iNo<<endl;
        }

        int SumFactors()
		{
			int iCnt = 0;
			int iSum = 0;
			
			if(iNo < 0)
            {
                iNo = -iNo;
            }
			
			for(iCnt = 1; iCnt < iNo/2; iCnt++)
			{
				if((iNo % iCnt) == 0)
				{
					iSum = iSum + iCnt;
				}
			}
			return iSum;
		}
};		

int main()
{
    Number nobj;
	int iRet = 0;

    nobj.Accept();
    nobj.Display();

    iRet = nobj.SumFactors();
	cout<<iRet<<endl;

    return 0;
}

File: PN11.java
import java.lang.*;
import java.util.*;

class Number
{
    private int iNo;

    public void Accept()
    {
        Scanner sobj = new Scanner(System.in);
        System.out.println("Enter number : ");
        this.iNo = sobj.nextInt();
    }

    public void Display()
    {
        System.out.println("Value is : "+this.iNo);
    }

    public void DisFactors()
    {
            int iCnt = 0;
            
			if(iNo < 0)
			{
				iNo = -iNo;
			}
			
            for(iCnt = 1; iCnt < iNo; iCnt++)
            {
                if((iNo % iCnt) == 0)
				{
					System.out.println(iCnt);
				}					
            }
            
    }
}

class PN11
{
    public static void main(String b[])
    {
            Number nobj = new Number();
            

            nobj.Accept();
            nobj.Display();

            nobj.DisFactors();
            
    }
}